roles/migrationcontroller/tasks/mcg.yml

```diff
@@ -0,0 +1,101 @@
+    k8s_facts: 
+      api_version: "noobaa.io/v1alpha1"
+      kind: NooBaa
+    register: noobaa_system
+
+  - set_fact:
+      noobaa_system_found: "{{ noobaa_system.resources|length > 0 }}"
+
+  - set_fact:
+      noobaa_system_namespace: "{{ mig_namespace if not noobaa_system_found else noobaa_system.resources[0].metadata.namespace }}"
+
+  - name: Creating a new NooBaa system
+    when: not noobaa_system_found|bool
+    k8s:
+     state: present
+     definition: "{{ lookup('template', 'noobaa_system.yml.j2') }}"
+
+# discover routes and secrets
+- block:
+  - name: Discovering NooBaa S3 endpoint
+    k8s_facts:
+      api_version: v1
+      kind: Service
+      name: s3
+      namespace: "{{ noobaa_system_namespace }}"
+    register: noobaa_s3_endpoint
+    retries: 10
+    delay: 3
+    until: noobaa_s3_endpoint.resources is defined and 
+           noobaa_s3_endpoint.resources[0] is defined and 
+           noobaa_s3_endpoint.resources[0].status.loadBalancer is defined and
+           noobaa_s3_endpoint.resources[0].status.loadBalancer.ingress is defined
+
+  - set_fact:
+      noobaa_endpoint: "{{ noobaa_s3_endpoint.resources[0].status.loadBalancer.ingress[0].hostname }}"
+
+  - name: Discovering NooBaa S3 Credentials
+    k8s_facts:
+      api_version: v1
+      kind: Secret
+      name: "{{ noobaa_admin_secret }}"
+      namespace: "{{ noobaa_system_namespace }}"
+    register: noobaa_admin_secret
+    retries: 20
+    delay: 3
+    until: noobaa_admin_secret.resources|length > 0
+
+  - set_fact:
+      noobaa_s3_access_key_id: "{{ noobaa_admin_secret.resources[0].data.AWS_ACCESS_KEY_ID | b64decode }}"
+      noobaa_s3_secret_access_key: "{{ noobaa_admin_secret.resources[0].data.AWS_SECRET_ACCESS_KEY | b64decode }}"
+
+  - name: Discovering NooBaa CA Bundle
+    shell: "{{ item }}"
+    loop:
+    - openssl s_client -showcerts -verify 5 -connect {{ noobaa_endpoint }}:443 < /dev/null | awk '/BEGIN/,/END/{ if(/BEGIN/){a++}; out="cert"a".crt"; print >out}' && for cert in *.crt; do newname=$(openssl x509 -noout -subject -in $cert | sed -n 's/^.*CN=\(.*\)$/\1/; s/[ ,.*]/_/g; s/__/_/g; s/^_//g;p').pem; mv $cert $newname; done;
+    - cat *.pem
+    - rm *.pem
+    register: noobaa_ca_bundle_raw
+    when: noobaa_s3_endpoint_proto == 'https'
+
+  - set_fact:
+      noobaa_ca_bundle: "{{ noobaa_ca_bundle_raw.results[1].stdout }}"
+    when: noobaa_s3_endpoint_proto == 'https'
+
+# NooBaa Bucket + migstorage creation
+- environment:
+    AWS_ACCESS_KEY_ID: "{{ noobaa_s3_access_key_id }}"
+    AWS_SECRET_ACCESS_KEY: "{{ noobaa_s3_secret_access_key }}"
+  block:
+  - name: Creating bucket for migstorage
+    aws_s3: 
+      bucket: "{{ noobaa_mig_bucket_name }}"
+      mode: create
+      s3_url: "{{ noobaa_s3_endpoint_proto }}://{{ noobaa_endpoint }}/"
+      validate_certs: false
+    register: bucket_status
+    retries: 20
+    delay: 3
+    until: not bucket_status.failed 
+
+  - name: Creating NooBaa backed migstorage 
```

Comment:
If we add the `when: migration_controller|bool` condition here we just don't create the mig-storage resource when it doesn't make sense while still allowing for the install of NooBaa.